pulseguardian/guardian.py

```diff
@@ -85,19 +91,17 @@ def clear_deleted_queues(self, queues):
 
         # Clean up bindings on queues that are not deleted.
         for queue_name in alive_queues_names:
-            self.clear_deleted_bindings(queue_name)
+            bindings = self.get_queue_bindings(all_bindings, queue_name)
+            self.clear_deleted_bindings(queue_name, bindings)
 
         db_session.commit()
 
-    def clear_deleted_bindings(self, queue_name):
-        rabbit_bindings = pulse_management.queue_bindings(config.rabbit_vhost,
-                                                          queue_name)
-
+    def clear_deleted_bindings(self, queue_name, all_bindings):
```

Comment:
I believe the second parameter should be named `queue_bindings`, since you are filtering for just this queue on line 94.